test/Microsoft.Sbom.Tool.Tests/IntegrationTests.cs

```diff
@@ -0,0 +1,219 @@
+
+        var (stdout, stderr, exitCode) = LaunchAndCaptureOutput(arguments);
+
+        Assert.AreEqual(stderr, string.Empty);
+        Assert.AreEqual(0, exitCode.Value);
+        Assert.IsTrue(File.Exists(outputFile), $"{outputFile} should have been created during validation");
+        Assert.IsTrue(File.ReadAllText(outputFile).Contains("\"Result\":\"Success\"", StringComparison.OrdinalIgnoreCase));
+    }
+
+    [TestMethod]
+    public void E2E_GenerateAndRedactManifest_RedactedFileIsSmaller_ReturnsZeroExitCode()
+    {
+        if (!IsWindows)
+        {
+            Assert.Inconclusive("This test is not (yet) supported on non-Windows platforms.");
+            return;
+        }
+
+        var testFolderPath = CreateTestFolder();
+        GenerateManifestAndValidateSuccess(testFolderPath);
+
+        var outputFolder = Path.Combine(testContext.ResultsDirectory, testContext.TestName, "redacted");
+        var originalManifestFolderPath = AppendFullManifestFolderPath(testFolderPath);
+        var originalManifestFilePath = Path.Combine(AppendFullManifestFolderPath(testFolderPath), ManifestFileName);
+        var arguments = $"redact -sp \"{originalManifestFilePath}\" -o \"{outputFolder}\" -verbosity verbose";
+
+        var (stdout, stderr, exitCode) = LaunchAndCaptureOutput(arguments);
+
+        Assert.AreEqual(stderr, string.Empty);
+        Assert.AreEqual(0, exitCode.Value);
+        Assert.IsTrue(stdout.Contains("Result=Success", StringComparison.OrdinalIgnoreCase));
+        var redactedManifestFilePath = Path.Combine(outputFolder, ManifestFileName);
+        var originalManifestSize = File.ReadAllText(originalManifestFilePath).Length;
+        var redactedManifestSize = File.ReadAllText(redactedManifestFilePath).Length;
+        Assert.IsTrue(redactedManifestSize > 0, "Redacted file must not be empty");
+        Assert.IsTrue(redactedManifestSize < originalManifestSize, "Redacted file must be smaller than the original");
+    }
+
+    private void GenerateManifestAndValidateSuccess(string testFolderPath)
+    {
+        var arguments = $"generate -ps IntegrationTests -pn IntegrationTests -pv 1.2.3 -m \"{testFolderPath}\"  -b . -bc \"{GetSolutionFolderPath()}\"";
+
+        var (stdout, stderr, exitCode) = LaunchAndCaptureOutput(arguments);
+
+        Assert.AreEqual(stderr, string.Empty);
+        var manifestFolderPath = AppendFullManifestFolderPath(testFolderPath);
+        var jsonFilePath = Path.Combine(manifestFolderPath, ManifestFileName);
+        var shaFilePath = Path.Combine(manifestFolderPath, "manifest.spdx.json.sha256");
+        Assert.IsTrue(File.Exists(jsonFilePath));
+        Assert.IsTrue(File.Exists(shaFilePath));
+        Assert.AreEqual(0, exitCode.Value);
+    }
+
+    private string CreateTestFolder()
+    {
+        var testFolderPath = Path.GetFullPath(Path.Combine(testContext.ResultsDirectory, testContext.TestName));
+        Directory.CreateDirectory(testFolderPath);
```

Comment:
Can we add a test cleanup method to delete these test dirs once the tests finish?